pkg/scheduler/plugins/reservation/plugin.go

```diff
@@ -474,6 +479,14 @@ func (pl *Plugin) filterWithReservations(ctx context.Context, cycleState *framew
 	allInsufficientResourcesByNode := sets.NewString()
 	var allInsufficientResourceReasonsByReservation []string
 	for _, rInfo := range matchedReservations {
+		// Before nominating a reservation in PreScore or Reserve, check the reservation by multiple plugins to make
+		// the Filter phase give a more accurate result. It is extensible to support more policies.
+		status := extender.RunReservationFilterPlugins(ctx, cycleState, pod, rInfo, nodeInfo)
```

Comment:
Will we implement different RunReservationFilterPlugins in different scenarios? Because I see all currently inplementation just return nil.

Reply:
As described in the "Special Notes" section, the existing FilterReservation plugins will be refactored into the re-defined method. Currently, we leave the methods for internal extensions and future usage.